x-pack/plugins/infra/server/routes/snapshot/lib/apply_metadata_to_last_path.ts

```diff
@@ -20,7 +20,7 @@ import { InfraSource } from '../../../lib/sources';
 export const isIPv4 = (subject: string) => /^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/.test(subject);
 
 type RowWithMetadata = MetricsAPIRow & {
-  [META_KEY]: object[];
+  [META_KEY]: Array<Record<string, string[] | number[]>>;
```

Comment:
```suggestion
  [META_KEY]: Array<Record<string, string | number>>;
```

The response should be something like:
`
  "__metadata__": [
    {
      "kubernetes.pod.name": "h2-p4-name__SS",
      "kubernetes.pod.ip": "10.136.197.76"
    }
  ]
`
so i'm not seeing where the arrays come in?

Reply:
Yes, you are right! I was "inspired" by @dgieselaar PR.
I managed to fix the typing and I think we need the arrays because the IP field can be an array, not sure why it can be also `number[]`

Reply:
I don't really know what this is about 😄 . Anything I can clarify?

Reply:
@dgieselaar I don't think so 😅 
I just used some of your changes from [here](https://github.com/dgieselaar/kibana/commit/069c4be6fff95a6e8f3571309f506ffe5d4000cc) in this PR and I got some type errors 🙈

Reply:
Thanks! I changed the type to what you suggested.